src/jlgen.jl

```diff
@@ -243,6 +243,17 @@ Core.Compiler.method_table(interp::GPUInterpreter, sv::InferenceState) =
     WorldOverlayMethodTable(interp.world)
 end
 
+# semi-concrete interepretation is broken with overlays (JuliaLang/julia#47349)
+@static if VERSION >= v"1.9.0-DEV.1248"
```

Comment:
We can change this version check to something like `VERSION >= v"1.9.0-DEV.1248" && VERSION <= v"1.9.0-DEV.XXXX"` once we fix https://github.com/JuliaLang/julia/issues/47349 in upstream.